framework/include/utils/KDTree.h

```diff
@@ -23,16 +23,16 @@ class KDTree
 
   virtual ~KDTree() = default;
 
-  void neighborSearch(Point & query_point,
+  void neighborSearch(const Point & query_point,
```

Comment:
:+1: on these.  Even if the rest of the PR has some problem it'd be worth breaking out and committing this API change alone.

I can't believe nobody tried to do something as simple as `neighborSearch(elem->centroid(), ...` before and noticed the compiler screaming about the attempt to pass a temporary to a non-const reference parameter.